### ISSUE TITLE
How do I access the responseheaders after a call to the api?

### ISSUE THREAD

**Issue by plamber** [microsoftgraph/msgraph-sdk-javascript#464]
Hello,
we are using the client to [copy a DriveItem](https://docs.microsoft.com/en-us/graph/api/driveitem-copy?view=graph-rest-1.0&tabs=javascript) from one Drive to another. 

```
const path = `${sourceDriveId}/items/${sourceFileId}/copy`;
return await client.api(path).post(
    {
        parentReference: {
            driveId: targetDriveId,
            id: targetFile
        },
        name: fileName
    }
```

To see if a file has been copied successfully and to receive the new driveItemId for other actions I have to follow [these instructions](https://docs.microsoft.com/en-us/graph/long-running-actions-overview?tabs=http).

In a nutshell I have to check the header from the response and perform additional checks until I get the necessary information.

Now to my questions:
- Is there an "easy" way today to get the driveItemId of a copy operation without doing all the activities above?
- Even if this is not supported by the SDK I am wondering if you have a code sample that shows me how to retrieve the "location" header property of the http response after executing the copy operation. I haven't found anything right now  
[AB#9766](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9766)

**Comment by nikithauc** [comment]
@plamber 


Please try explicitly setting the response type to RAW as follows -
```
 client.api('/me/drive/items/{item-id}/copy').responseType(ResponseType.RAW)
      .post(driveItem)
          .then((res) => {
        console.log(res.headers);
    })
    .catch((err) => {
        console.log(err);
    }); 
```

The headers should be available from the raw response. 

Please let me know if this works.

**Comment by plamber** [comment]
Hi @nikithauc,
thank you. Will give it a try but might be able to give you feedback in a couple of days only.

Cheers.

**Comment by plamber** [comment]
Hi @nikithauc,
thank you. I was able to successfully retrieve the location header with this code.

```
client.api('/me/drive/items/{item-id}/copy').responseType(ResponseType.RAW)
      .post(driveItem)
          .then((res) => {
        let location = res.headers.get("location");
    })
    .catch((err) => {
        console.log(err);
    }); 
```

I would recommend to update the JavaScript documentation [here](https://docs.microsoft.com/en-us/graph/long-running-actions-overview?tabs=javascript) to show how you can retrieve the location header.

I think you can also remove the part where '.version('beta')' is used. Apparently this is no more required.

Thank you
